diff --git a/dist/get-placement.js b/dist/get-placement.js
index 098ceeff218a208f43ebb99a00943651494d2bcb..ee31b9aa90cd9b536652ed37fba130eeaf4b665b 100644
--- a/dist/get-placement.js
+++ b/dist/get-placement.js
@@ -77,22 +77,26 @@ function getOffsetMiddleware(arrowElement, opts) {
 }
 function getFlipMiddleware(opts) {
   if (!opts.flip) return;
-  const boundary = resolveBoundaryOption(opts.boundary);
-  return (0, import_dom.flip)({
-    ...boundary ? { boundary } : void 0,
-    padding: opts.overflowPadding,
-    fallbackPlacements: opts.flip === true ? void 0 : opts.flip
+  return (0, import_dom.flip)(() => {
+    const boundary = resolveBoundaryOption(opts.boundary);
+    return {
+      ...boundary ? { boundary } : void 0,
+      padding: opts.overflowPadding,
+      fallbackPlacements: opts.flip === true ? void 0 : opts.flip
+    };
   });
 }
 function getShiftMiddleware(opts) {
   if (!opts.slide && !opts.overlap) return;
-  const boundary = resolveBoundaryOption(opts.boundary);
-  return (0, import_dom.shift)({
-    ...boundary ? { boundary } : void 0,
-    mainAxis: opts.slide,
-    crossAxis: opts.overlap,
-    padding: opts.overflowPadding,
-    limiter: (0, import_dom.limitShift)()
+  return (0, import_dom.shift)(() => {
+    const boundary = resolveBoundaryOption(opts.boundary);
+    return {
+      ...boundary ? { boundary } : void 0,
+      mainAxis: opts.slide,
+      crossAxis: opts.overlap,
+      padding: opts.overflowPadding,
+      limiter: (0, import_dom.limitShift)()
+    };
   });
 }
 function getSizeMiddleware(opts) {
diff --git a/dist/get-placement.mjs b/dist/get-placement.mjs
index 23f789533bcd8d1237bd659dc70ca6504c6fa385..06bba8e9c067adafb3b23e22ecfa51c4d828f57f 100644
--- a/dist/get-placement.mjs
+++ b/dist/get-placement.mjs
@@ -53,22 +53,26 @@ function getOffsetMiddleware(arrowElement, opts) {
 }
 function getFlipMiddleware(opts) {
   if (!opts.flip) return;
-  const boundary = resolveBoundaryOption(opts.boundary);
-  return flip({
-    ...boundary ? { boundary } : void 0,
-    padding: opts.overflowPadding,
-    fallbackPlacements: opts.flip === true ? void 0 : opts.flip
+  return flip(() => {
+    const boundary = resolveBoundaryOption(opts.boundary);
+    return {
+      ...boundary ? { boundary } : void 0,
+      padding: opts.overflowPadding,
+      fallbackPlacements: opts.flip === true ? void 0 : opts.flip
+    };
   });
 }
 function getShiftMiddleware(opts) {
   if (!opts.slide && !opts.overlap) return;
-  const boundary = resolveBoundaryOption(opts.boundary);
-  return shift({
-    ...boundary ? { boundary } : void 0,
-    mainAxis: opts.slide,
-    crossAxis: opts.overlap,
-    padding: opts.overflowPadding,
-    limiter: limitShift()
+  return shift(() => {
+    const boundary = resolveBoundaryOption(opts.boundary);
+    return {
+      ...boundary ? { boundary } : void 0,
+      mainAxis: opts.slide,
+      crossAxis: opts.overlap,
+      padding: opts.overflowPadding,
+      limiter: limitShift()
+    };
   });
 }
 function getSizeMiddleware(opts) {
